### PR TITLE
Fixed a corner case for IDL union code generation in Java

### PR DESCRIPTION
**Pull request by mitza-oci**
Unions with implicit defaults that don't have a branch selected need their discriminator values copied from C++ to Java.
There may be one more issue with unions that I'll add to this PR, but I wanted to get CI going on this part of it.

### CODE REVIEW

File: java/idl2jni/codegen/im_jni.cpp [hunk 1]
```diff
@@ -1791,6 +1795,30 @@ bool idl_mapping_jni::gen_union(UTL_ScopedName *name,
       "  " + disc_ty + " jdisc = source._d ();\n";
   }
 
+  if (!hasDefault && default_value.computed_ != 0) {
```
Comment (jrw972) [inline java/idl2jni/codegen/im_jni.cpp:1798]: Why is hasDefault negated?

Comment (mitza-oci) [reply java/idl2jni/codegen/im_jni.cpp:1798]: This is from the IDL-to-Java spec (IDL-to-C++ is very similar):
> Two default modifier methods, both named __default(), are generated if there is no explicit default case label, and the set of case labels does not completely cover the possible values of the discriminant.

Comment (jrw972) [reply java/idl2jni/codegen/im_jni.cpp:1798]: So I'm assuming that `default_value.computed_ != 0` is a way of detecting if the case labels don't completely cover the values of the discriminant?

Comment (mitza-oci) [reply java/idl2jni/codegen/im_jni.cpp:1798]: https://github.com/DOCGroup/ACE_TAO/blob/7c298ab69026c64c027d63530f7ca275f0bc9c73/TAO/TAO_IDL/include/ast_union.h#L122

**Pull request merged**